### DISCUSSION TITLE
installation db error

### DISCUSSION THREAD

**Discussion by J0han3s** [firefly-iii/firefly-iii#5495] [Firefly III]
**Hi I am trying to install using docker, but getting the below error on my db** 

                                                                           ,
  SQLSTATE[HY000] [1045] Access denied for user 'firefly'@'192.168.80.3' (usi  ,
  ng password: YES) (SQL: insert into `oauth_clients` (`user_id`, `name`, `se  ,
  cret`, `provider`, `redirect`, `personal_access_client`, `password_client`,  ,
   `revoked`, `updated_at`, `created_at`) values (?, FireflyIII Personal Acce  ,
  ss Client, hmjIdDqJpD0dYstjUE8s9yTcFMd4ENztZJw1wf9C, ?, http://localhost, 1  ,
  , 0, 0, 2022-01-05 23:34:17, 2022-01-05 23:34:17))                           ,
                                                                               ,
,
In Connector.php line 70:,
                                                                               ,
  SQLSTATE[HY000] [1045] Access denied for user 'firefly'@'192.168.80.3' (usi  ,
  ng password: YES)                                                            ,
                                                                               ,
,
Updated version.,
+------------------------------------------------------------------------------+,
|                                                                              |,
| Thank you for installing Firefly III, v5.6.9!                                |,
|                                                                              |,
|                                                                              |,
+------------------------------------------------------------------------------+,
Go!,
AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 192.168.80.3. Set the 'ServerName' directive globally to suppress this message,
AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 192.168.80.3. Set the 'ServerName' directive globally to suppress this message,
[Wed Jan 05 23:34:19.399331 2022] [mpm_prefork:notice] [pid 1] AH00163: Apache/2.4.38 (Debian) configured -- resuming normal operations,
[Wed Jan 05 23:34:19.399724 2022] [core:notice] [pid 1] AH00094: Command line: 'apache2 -D FOREGROUND'

**and the below on my app - I used the default installation and only changed the port as port 80 is already used on my setup**

`
2022-01-05 22:33:56 4 [Warning] Access denied for user 'firefly'@'192.168.80.3' (using password: YES),
Version: '10.6.5-MariaDB-1:10.6.5+maria~focal'  socket: '/run/mysqld/mysqld.sock'  port: 3306  mariadb.org binary distribution,
2022-01-05 22:33:51 3 [Warning] Aborted connection 3 to db: 'unconnected' user: 'unauthenticated' host: '192.168.80.3' (This connection closed normally without authentication),
2022-01-05 22:33:14 0 [Warning] 'proxies_priv' entry '@% root@fireflyiiidb' ignored in --skip-name-resolve mode.,
2022-01-05 22:33:14 0 [Note] mariadbd: ready for connections.,
2022-01-05 22:33:14 0 [Note] Server socket created on IP: '::'.,
2022-01-05 22:33:14 0 [Note] Server socket created on IP: '0.0.0.0'.,
2022-01-05 22:33:14 0 [Warning] You need to use --log-bin to make --expire-logs-days or --binlog-expire-logs-seconds work.`

**Comment by JC5** [comment]
Did you change the password in the .env file?

**Reply by J0han3s** [reply]
hello,

thanks for the prompt response. No kept it unchanged in both .environmental and docket-compose. wanted to make sure i understand the installation process first, before chaning anything. 

only thing changes is port from host.

Any advice?

**Comment by J0han3s** [comment]
hello,

thanks for the prompt response. No kept it unchanged in both .environmental and docket-compose

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by J0han3s** [reply]
I've had a play around... if i change any of the database details ( username. db name, password) , the installation has the above error. However, if i use the default details it works.

This is eventhough i am changing the details in both docker-compose and in .env file.

**Reply by JC5** [reply]
Once its created, the password can't be changed in the database (easily) unless you recreate the volume from scratch. The password is set the first time it starts, after that it cant be easily changed anymore. 

So you launched it the first time with the original values, and after that they stuck to the DB.

**Comment by J0han3s** [comment]
hI James,
i am using Docker, so tried the differenr passwords because starting each docker instances , before deleting the files and instance and starting another docker instance.

can the password contain special characters, numbers and caps ??

thanks

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by JC5** [reply]
Yes it can, but I know people have had trouble with # and " in environment files like `.env` files.

**Comment by J0han3s** [comment]
thanks, is this the same for db_username ?

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by JC5** [reply]
Do you want special chars in the username? Try to keep it without special chars, lowercare

**Comment by J0han3s** [comment]
thank you, this has now worked.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
